src/panels/config/integrations/integration-panels/zha/dialog-zha-device-children.ts

```diff
@@ -0,0 +1,130 @@
+import {
+  CSSResult,
+  customElement,
+  html,
+  internalProperty,
+  LitElement,
+  property,
+  TemplateResult,
+} from "lit-element";
+import memoizeOne from "memoize-one";
+import { computeRTLDirection } from "../../../../../common/util/compute_rtl";
+import "../../../../../components/ha-code-editor";
+import { createCloseHeading } from "../../../../../components/ha-dialog";
+import { haStyleDialog } from "../../../../../resources/styles";
+import { HomeAssistant } from "../../../../../types";
+import { ZHADeviceChildrenDialogParams } from "./show-dialog-zha-device-children";
+import "../../../../../components/data-table/ha-data-table";
+import type {
+  DataTableColumnContainer,
+  DataTableRowData,
+} from "../../../../../components/data-table/ha-data-table";
+import { fetchDevices, ZHADevice } from "../../../../../data/zha";
+
+export interface DeviceRowData extends DataTableRowData {
+  id: string;
+  name: string;
+  lqi: number;
+}
+
+@customElement("dialog-zha-device-children")
+class DialogZHADeviceChildren extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @internalProperty() private _device: any;
+
+  @internalProperty() private _devices: Map<string, ZHADevice> = new Map();
+
+  private _deviceChildren = memoizeOne(
+    (device: ZHADevice, devices: Map<string, ZHADevice>) => {
+      const outputDevices: DeviceRowData[] = [];
+
+      device.neighbors.forEach((child) => {
+        const zhaDevice: ZHADevice | undefined = devices.get(child.ieee);
+        if (zhaDevice) {
+          outputDevices.push({
+            name: zhaDevice.user_given_name || zhaDevice.name,
+            id: zhaDevice.device_reg_id,
+            lqi: child.lqi,
+          });
+        }
+      });
+
+      return outputDevices;
+    }
+  );
+
+  private _columns: DataTableColumnContainer = {
+    name: {
+      title: "Name",
+      sortable: true,
+      filterable: true,
+      direction: "asc",
+      grows: true,
+    },
+    lqi: {
+      title: "LQI",
+      sortable: true,
+      filterable: true,
+      direction: "asc",
+      width: "75px",
+    },
+  };
+
+  public async showDialog(
+    params: ZHADeviceChildrenDialogParams
+  ): Promise<void> {
+    this._device = params.device;
+    await this._fetchData();
```

Comment:
Don't await this, and show a loader in the dialog until `this._devices` is loaded.

Reply:
Is there another example of this that I can look at?

Reply:
Just remove that await and add this in render:
```ts
${!this._devices ? html`<ha-circular-progress alt="Loading" size="large" active>
              </ha-circular-progress>` : html`<ha-data-table></ha-data-table>`}
              ```